Fix/data import (#1412)
hadley/r4ds: data-import.qmd

In the `favourite.food` column, there are a bunch of food items, and then the character string `N/A`, which should have been a real `NA` that R will recognize as "not available".
This is something we can address using the `na` argument.

> By default `read_csv()` only recognizes the empty string (`""`) and the character string `"NA"` as `NA`s, we want it to also recognize the character string `"N/A"`.

Reviewer: ```suggestion
By default `read_csv()` only recognizes empty strings (`""`) in this dataset as `NA`s, we want it to also recognize the character string `"N/A"`.
```

```{r}
#| message: false

> students <- read_csv("data/students.csv", na = c("N/A", "", "NA"))

Reviewer: ```suggestion
students <- read_csv("data/students.csv", na = c("N/A", ""))
```

-   `col_integer()` reads integers. We seldom distinguish integers and doubles in this book because they're functionally equivalent, but reading integers explicitly can occasionally be useful because they occupy half the memory of doubles.
-   `col_character()` reads strings. This is sometimes useful to specify explicitly when you have a column that is a numeric identifier, i.e. long series of digits that identifies some object, but it doesn't make sense to (e.g.) divide it in half, for example, a phone number, social security number, credit card number, etc.
-   `col_factor()`, `col_date()`, and `col_datetime()` create factors, dates, and date-times respectively; you'll learn more about those when we get to those data types in @sec-factors and @sec-dates-and-times.
-   `col_number()` is a permissive numeric parser that will ignore non-numeric components, and is particularly useful for currencies. You'll learn more about it in @sec-numbers.
-   `col_skip()` skips a column so it's not included in the result, which can be useful for speeding up reading the data if you have a large CSV file and you only want to use some of the columns.

> It's possible to override the default column by switching from `list()` to `cols()` and specifying `.default`:

Reviewer: ```suggestion
It's also possible to override the default column by switching from `list()` to `cols()` and specifying `.default`:
```

-   `col_integer()` reads integers. We seldom distinguish integers and doubles in this book because they're functionally equivalent, but reading integers explicitly can occasionally be useful because they occupy half the memory of doubles.
-   `col_character()` reads strings. This is sometimes useful to specify explicitly when you have a column that is a numeric identifier, i.e. long series of digits that identifies some object, but it doesn't make sense to (e.g.) divide it in half, for example, a phone number, social security number, credit card number, etc.
-   `col_factor()`, `col_date()`, and `col_datetime()` create factors, dates, and date-times respectively; you'll learn more about those when we get to those data types in @sec-factors and @sec-dates-and-times.
-   `col_number()` is a permissive numeric parser that will ignore non-numeric components, and is particularly useful for currencies. You'll learn more about it in @sec-numbers.
-   `col_skip()` skips a column so it's not included in the result, which can be useful for speeding up reading the data if you have a large CSV file and you only want to use some of the columns.

> It's possible to override the default column by specifying `.default`.
> You can also subsitute `cols()` for `list()`:

Reviewer: ```suggestion
It's possible to override the default column by switching from `list()` to `cols()` and specifying `.default`:
```
